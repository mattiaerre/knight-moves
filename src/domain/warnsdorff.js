import makeAllowedMoves from './makeAllowedMoves';

function warnsdorff(position) {
  const allowedMoves = makeAllowedMoves(position);

  return allowedMoves
    .map((move) => [...move, makeAllowedMoves(move).length - 1])
    .sort((a, b) => {
      if (a[2] < b[2]) {
        return -1;
      }
      if (a[2] > b[2]) {
        return 1;
      }
      return -1;
    });
}

export default warnsdorff;
