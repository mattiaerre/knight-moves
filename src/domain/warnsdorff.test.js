import warnsdorff from './warnsdorff';

test("Warnsdorff's rule", () => {
  const moves = warnsdorff(['b', 4]);
  expect(moves).toMatchSnapshot();
});
