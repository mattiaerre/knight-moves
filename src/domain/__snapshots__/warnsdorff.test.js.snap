// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warnsdorff's rule 1`] = `
Array [
  Array [
    "a",
    2,
    2,
  ],
  Array [
    "a",
    6,
    3,
  ],
  Array [
    "c",
    2,
    5,
  ],
  Array [
    "c",
    6,
    7,
  ],
  Array [
    "d",
    5,
    7,
  ],
  Array [
    "d",
    3,
    7,
  ],
]
`;
